server/index.js

```diff
@@ -4,18 +4,81 @@ const db = require('../database/index.js');
 const app = express();
 const PORT = 5005;
 
-app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.urlencoded({ extended: true }));
 app.use(bodyParser.json());
 app.use('/rooms/:id', express.static(__dirname + '/../client/dist'));
 app.use('/rooms/:id', express.static(__dirname + '/../public'));
 
-app.get('/rooms/:id/getPhotosByRoomID', (req, res) => {
-  db.getPhotosByRoomId(req.params.id, (err, photos) => {
-    if (err) {
-      res.status(500).send(err);
+app.get('/rooms/:id/getPhotosByRoomId', (req, res) => {
+  db.getPhotosByRoomId(req.params.id)
+    .then(response => {
+      if (response.length) {
+        res.status(200).send(response);
+      } else {
+        res.status(400).send('No photos associated with room');
+      }
+    })
+    .catch(error => {
+      res.status(500).send(error);
+    });
+});
+
+app.post('/rooms/addRoomPhoto/', (req, res) => {
+
+  const expectedShape = {
+    room_id: 'Number',
+    name: 'String',
+    caption: 'String',
+    storage_url: 'String'
+  };
+
+  for (let key in expectedShape) {
+    if (!req.body[key]) {
+      res.status(500).send(`missing data ${key}`);
     }
-    res.status(200).send(photos);
-  });
+  }
+
+  db.addRoomPhotos(req.body)
+    .then(response => {
+      console.log(response);
+    })
+    .catch(error => {
+      res.status(500).send(error);
+    });
+});
+
+app.put('/rooms/updateRoomPhoto', (req, res) => {
+  const { room_id, _id } = req.body;
+
+  if (!room_id || !_id) {
+    res.status(500).send('Must specificy both room_id and object_id');
+  } else {
+    db.updateRoomPhoto(req.body)
+      .then(response => {
+        res.status(200).send('Room Updated');
+      })
+      .catch(error => {
+        res.status(500).send('Unable to update room');
+      });
+  }
+});
+
+app.delete('/rooms/deleteRoom', (req, res) => {
+
+  const { room_id } = req.body;
+
+  if (!room_id) {
+    res.status(500).send('ID does not match record in database');

```

Comment:
I need to send message that notifies instead of just sending the status code. Didn't even know you could do this. Thanks!